src/Controls/src/Core/Handlers/Items/iOS/SelectableItemsViewController.cs

```diff
@@ -57,6 +57,26 @@ internal void SelectItem(object selectedItem)
 				// Ensure the selected index is updated after the collection view's items generation is completed
 				CollectionView.PerformBatchUpdates(null, _ =>
 				{
+					// Exit if the ItemsSource reference no longer matches the one captured at invocation.
+					if (!ReferenceEquals(ItemsView.ItemsSource, originalSource))
+					{
+						return;
+					}
+
+					// Recalculate the index for the selectedItem now that the collection may have changed.(Adding, deleting etc..)
+					var updatedIndex = GetIndexForItem(selectedItem);
```

Comment:
I think we need an additional check in here to see if `DisposeItemsSource();` has been called

I'm working on getting
https://github.com/dotnet/maui/pull/29281

Green on the net10 branch and I'm noticing that the navigation to CollectionView Gallery is crashing

The fix on this PR almost works but the problem is is that the handlers gets "Disconnected"

<img width="1167" alt="image" src="https://github.com/user-attachments/assets/fe59871a-a514-4dd4-97b5-b09a9c13fdf5" />

Which sets the ItemSource property on the Controller to an EmptySource

Which causes this line to crash

Which I think would also be fixed by your round trip fixes :-/